src/OpenTelemetry/Logs/Builder/LoggerProviderBuilderBase.cs

```diff
@@ -91,6 +93,14 @@ internal LoggerProvider Build()
         bool validateScopes = false;
 #endif
         var serviceProvider = services.BuildServiceProvider(validateScopes);
+        var configuration = serviceProvider.GetRequiredService<IConfiguration>();
+        configuration.TryGetStringValue(SdkConfigDefinitions.SdkDisableEnvVarName, out var envVarValue);
+
+        if (bool.TryParse(envVarValue, out bool result) && result)
+        {
+            serviceProvider.Dispose();
+            return new NoopLoggerProvider();
```

Comment:
2 comments:

1) Would probably be good to add a log here. Don't want support tickets raised for missing telemetry when users have shot themselves in the foot 😄

2) This won't work for hosting scenarios. See line 80. This code is only invoked for manual/detached bootstrap. Something else will need to be done for hosting style. Ideally we would have a single solution for both but I haven't looked at where that might go.